### ISSUE TITLE
Run appears to fail if Java Desktop API is not supported.

### ISSUE THREAD

**Issue by pfeyz** [fathominfo/processing-p5js-mode#17]
I'm running arch linux and a lightweight non-desktop window manager. 

When attempting to run a sketch from Processing 3 (0263) in p5.js mode I get the following error:

```
java.lang.UnsupportedOperationException: 
The BROWSE action is not supported on the current platform!
```

The error suppresses the "Server running at [port]"  [status message](https://github.com/fathominfo/processing-p5js-mode/blob/master/src/processing/mode/p5js/p5jsEditor.java#L447). It appears everything has failed when in fact the server is running and serving your sketch, you're just missing the convenience of a browser automatically opening the sketch at the url it's served at.

I was in the Information Design workshop (which was great) at the recent Processing Community Day and was distracted while trying to figure out what was wrong. 

I suspect that wrapping the call to [Desktop.getDesktop.browse()](https://github.com/fathominfo/processing-p5js-mode/blob/master/src/processing/mode/p5js/p5jsEditor.java#L450) in a test for Desktop API support
```
        if(Desktop.isDesktopSupported()){
            Desktop.getDesktop().browse(new URI(server.getAddress()));
        } else {
           ... tell user to navigate browser to server's address ...
        }
```
would be enough, as no error would be triggered and you'd see the status message telling you which port to navigate your browser to.

 I'm not familiar enough with the build process here to test this myself, though if pointed in the right direction I'd be happy to.

**Comment by benfry** [comment]
Hm, though can you run a web browser on your system? If not, then you won't be able to do much with the Mode. Or if you can, any ideas on why `Desktop` wouldn't work properly?

**Comment by pfeyz** [comment]
Yes, I can run a web browser. 

[This quotes](https://stackoverflow.com/a/18004421) the Java documentation saying that on Linux the Desktop API depends on the Gnome libraries. I'm not running a Gnome-based desktop environment. 

I understand that my particular setup may be uncommon but I'm not sure if users with KDE or Xfce, other popular Gnome alternatives, would have the same experience I did.

**Comment by benfry** [comment]
Fixed for the next release (1.1). 

Problem was that cross-platform URL opening (i.e. the Linux-specific workaround set) was partly shut off in order to track down other URL launching issues on other platforms. Found an intermediate solution that should work for both scenarios.
